tests/trx_generator/trx_provider.cpp

```diff
@@ -73,15 +73,31 @@ namespace eosio::testing {
    }
 
    void p2p_connection::disconnect() {
-      ilog("Closing socket.");
-      _p2p_socket.close();
```

Comment:
Now that `_p2p_socket.close();` is removed, where is disconnect triggered?

Reply:
At destruction of `p2p_connection`.